kansha/app/comp.py

```diff
@@ -244,6 +245,7 @@ def set_config(self, config_filename, conf, error):
 
         # search_engine engine configuration
         self.search_engine = SearchEngine(**conf['search'])
+        fts_schema.Card.update_schema(self.card_extensions)
```

Comment:
> Le 12 janv. 2016 à 18:50, Romuald Texier-Marcadé <notifications@github.com> a écrit :
> 
> In kansha/app/comp.py:
> 
> > @@ -244,6 +245,7 @@ def set_config(self, config_filename, conf, error):
> >  
> >          # search_engine engine configuration
> >          self.search_engine = SearchEngine(**conf['search'])
> > +        fts_schema.Card.update_schema(self.card_extensions)
> 
> Proposed API:
> 
>    MyDocument = schema.Schema('MyDocument'
> )
>    MyDocument.add_field(
> 'title', schema.TEXT(stored=True
> ))
>    MyDocument.add_field(
> 'tags', schema.TEXT(stored=False
> ))
>    MyDocument.add_field(
> 'pages', schema.INT(stored=True
> ))
>    MyDocument.add_field(
> 'description'
> , schema.TEXT)
>    MyDocument.add_field(
> 'price', schema.FLOAT(stored=True, indexed=False))

Yes, that’s the idea.

With more syntaxic sugar. Something like:

  MyDocument = schema.Schema('MyDocument') +
    schema.Text('title', stored=True) +
    schema.Text('tags', stored=False) +
    schema.Integer('pages', stored=True) +
    schema.Text('description') +
    schema.Float('price', stored=True, indexed=False)

> Usage stay identical as with a declative schema.
> 
> doc1 =
>  MyDocument(
>             
> 'doc1', title=u'Titre', tags=u'Un livre français best seller'
> ,
>             
> pages=89, description=u'Description de qualité, avec des services.', price=10.0
> )
> query 
> = MyDocument.match(u'tests'
> )
> query 
> = MyDocument.tags.match(u'best'
> )
> query 
> = (MyDocument.pages == 89) |
>  MyDocument.description.match(
>             
> u'practices')
> —
> Reply to this email directly or view it on GitHub.
>